src/http.cpp

```diff
@@ -271,8 +268,8 @@ int Exiv2::http(Exiv2::Dictionary& request,Exiv2::Dictionary& response,std::stri
 
     ////////////////////////////////////
     // open the socket
-    int     sockfd = (int) socket(AF_INET , SOCK_STREAM,IPPROTO_TCP) ;
-    if (    sockfd < 0 ) return error("unable to create socket\n",NULL,NULL,0) ;
+    int     sockfd = socket(AF_INET , SOCK_STREAM,IPPROTO_TCP) ;
+    if (    sockfd < 0 ) return error(errors, "unable to create socket\n",NULL,NULL,0) ;
```

Comment:
I do not understand something here. Before we were passing as the first argument the **msg** string, and now we are passing the errors variable. Were this function used incorrectly before ?

Reply:
I believe the previous version was buggy. It was essentially formatting `msg` with the last three parameters and then calling `report(buffer, errors)` which is just `errors += buffer + '\n'` (a little simplified). However, `report` took `errors` by reference and returned nothing, so `error` looks like it should append the current error message to `errors`. However it was not doing that, as it took `errors` by value.

@clanmills Do you remember if this is the case? Git says you are the author.